workflows/trigger-lp-build.py

```diff
@@ -185,19 +189,22 @@ def main(argv):
 
         arch = release.getDistroArchSeries(archtag=build_arch)
         if series == "xenial":
+            if not snapcraft_channel:
+                snapcraft_channel = "4.x/stable"
             request = snap.requestBuild(archive=primary_archive,
-                                        channels={"snapcraft": "4.x/stable"},
+                                        channels={"snapcraft": snapcraft_channel},
                                         distro_arch_series=arch,
                                         pocket='Updates',
                                         snap_base='/+snap-bases/core')
         else:
-            if series == 'bionic':
-                # 6.0 onwards does not support i386, that we publish for core18
-                snapcraft_chan = "5.x/stable"
-            else:
-                snapcraft_chan = "latest/stable"
+            if not snapcraft_channel:
+                if series == 'bionic':
+                    # 6.0 onwards does not support i386, that we publish for core18
+                    snapcraft_channel = "5.x/stable"
+                else:
+                    snapcraft_channel = "latest/stable"
             request = snap.requestBuild(archive=primary_archive,
-                                        channels={"snapcraft": snapcraft_chan},
+                                        channels={"snapcraft": snapcraft_channel},
```

Comment:
Instead of having the decision logic around which channel to use split out here, we could determine this earlier in the function in one place. This would make it easier to maintain.  Another thing, does it make sense to allow overriding the snapcraft channel in the cases of bionic/xenial?